terminus/models/road.py

```diff
@@ -91,6 +92,33 @@ def reverse(self):
     def be_two_way(self):
         pass
 
+    def length(self, initial=0, final=None):
+        distances = self.get_waypoint_distances()
+        if final is None:
+            return math.fsum(distances[initial:])
+        return math.fsum(distances[initial:final])
+
+    def get_waypoint_positions(self):
+        waypoints = self.get_waypoints()
+        points = []
+        for waypoint in waypoints:
+            points.append(waypoint.center)
+        return points
+
+    def get_waypoint_distances(self):
```

Comment:
This is definitely not necessary, but just some food for thought if you want to get more into the "Pythonic" way of doing things:

- There is a `zip` function that you can use to, given two lists, create a list of tuples (so [1,2,3] and [4,5,6] becomes [(1,4), (2,5), (3,6)].
- There is the `map` function which you can use to convert the elements of a list into other things.

So the idea would be:
- zip the waypoint positions list with a copy of itself that doesn't have the first element. So if you have something like [(1,1), (2,2), (3,3)] you would get [((1,1),(2,2)), ((2,2),(3,3))].
- map that list by doing lambda pair_of_points: pair_of_points[0].distance_to(pair_of_points[1])

Again, not necessary, just a good exercise to think in a different way :)